Resources/Prototypes/Entities/Structures/Furniture/beds.yml

```diff
@@ -24,6 +20,14 @@
   - type: Strap
     position: Down
     rotation: -90
+  # Construction has to be before PlaceableSurface or you'll place things onto the entity instead of deconstructing it.
+  - type: Construction
+    graph: bed
+    node: bed
+  # So that you can put bedsheets on beds again. Would be cool to have a tag
+  # for this so that only bedsheets can be placed.
+  - type: PlaceableSurface
+    placeCentered: true
```

Comment:
uhhh this isnt how that should work? prototype component order shouldnt matter, are you sure this is whats happening

Reply:
I worked it out through trial and error. Before I moved it below Construction, only DogBed was deconstructable, after I moved it both were deconstructable. From looking at how tables work it seems all the used tables which are deconstructable are children of BaseTable, so I figured that this combined with the fact that DogBed worked while Bed didn't meant that something in the parenting process allows for Construction to override PlaceableSurface. Thus I moved Construction above PlaceableSurface and it worked, leading to my comment's conclusion.

I'm not sure what the actual code-wise cause of it is, but this is how I could get it working in practice.

Reply:
Event order moment @mirrorcult?

Reply:
i guess? construction queues events which fixed most of these issues i thoguht

Reply:
Potential fix PR up https://github.com/space-wizards/space-station-14/pull/6385 after lamarr mentioned it on discord.